src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,199 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node first;
+    private Node last;
+    private int size = 0;
+
     @Override
     public void add(T value) {
+        Node node = new Node<>(null, value, null);
+        if (size == 0) {
+            first = node;
+            last = node;
+        } else {
+            last.next = node;
+            node.prev = last;
+            last = node;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException();
+        }
+
+        Node node = new Node<>(null, value, null);
```

Comment:
Similar to the previous comment, ensure that the `Node` instantiation uses the generic type `T` for type safety.